### PR TITLE
Add validator for preference env var values to avoid typing os-dependent path separators.

### PR DESCRIPTION
**Pull request by lyuborusev**
The validator is going to be used in preferences options UI

### CODE REVIEW

File: source/MaterialXCore/Util.cpp [hunk 1]
```diff
@@ -62,6 +62,14 @@ bool isValidNamespace(const string& name)
     return isValidName(name) && name.find(':') == std::string::npos;
 }
 
+bool isValidEnvironmentPath(const string& path)
+{
+    // The column and semicolumn chars are used to separate paths in env vars on different os.
+    // Do not allow both of them to have more portable values.
+    return path.find(':') == std::string::npos
```
Comment (feldstj) [inline source/MaterialXCore/Util.cpp:69]: Shouldn't this be a different check based upon the OS?

File: source/MaterialXCore/Util.cpp [hunk 2]
```diff
@@ -62,6 +62,14 @@ bool isValidNamespace(const string& name)
     return isValidName(name) && name.find(':') == std::string::npos;
 }
 
+bool isValidEnvironmentPath(const string& path)
```
Comment (bernardkwok) [inline source/MaterialXCore/Util.cpp:65]: This belongs more as part of the Format module as a file utility.
- Also I don't see anything specific to "environment" but it's more a path
checker.
- Finally there are existing constants for path separators you can use.

**Review by bernardkwok — CHANGES_REQUESTED** [review]

File: source/MaterialXCore/Util.cpp (hunk 1 above)
Comment (lyuborusev) [reply source/MaterialXCore/Util.cpp:69]: I wanted to have checked against both separators to make the pref variable file more portable between different OS.

File: source/MaterialXCore/Util.cpp (hunk 2 above)
Comment (lyuborusev) [reply source/MaterialXCore/Util.cpp:65]: The naming was intended to point out that we are checking the Preference Environment Variables. I can improve the wording, or just leave isValidPath.

**Review by feldstj — APPROVED** [review]

**Pull request merged**